torch/jit/_monkeytype_config.py

```diff
@@ -0,0 +1,127 @@
+        def __init__(self, store: CallTraceStore):
+            super().__init__(store)
+
+        def log(self, trace: CallTrace) -> None:
+            self.traces.append(trace)
+
+    class JitTypeTraceStore(CallTraceStore):
+        def __init__(self):
+            super().__init__()
+            # A dictionary keeping all collected CallTrace
+            # key is fully qualified name of called function
+            # value is list of all CallTrace
+            self.trace_records: Dict[str, list] = defaultdict(list)
+
+        def add(self, traces: Iterable[CallTrace]):
+            for t in traces:
+                qualified_name = get_qualified_name(t.func)
+                self.trace_records[qualified_name].append(t)
+
+        def filter(
+            self,
+            qualified_name: str,
+            qualname_prefix: Optional[str] = None,
+            limit: int = 2000
+        ) -> List[CallTraceThunk]:
+            return self.trace_records[qualified_name]
+
+        def analyze(self, qualified_name: str) -> Dict:
+            # Analyze the types for the given module
+            # and create a dictionary of all the types
+            # for arguments and return values this module can take
+            records = self.trace_records[qualified_name]
+            all_args = defaultdict(set)  # type:  ignore[var-annotated]
+            for record in records:
+                for arg, arg_type in record.arg_types.items():
+                    all_args[arg].add(arg_type)
+                all_args["return_type"].add(record.return_type)
+            return all_args
+
+        def consolidate_types(self, qualified_name: str) -> Dict:
+            all_args = self.analyze(qualified_name)
+            # If there are more types for an argument,
+            # then consolidate the type to `Union` and replace the entry
+            # by type `Union`.
+            # This currently handles only cases with only one return value
+            # TODO: Consolidate types for multiple return values
+            for arg, types in all_args.items():
+                _all_type = " "
+                for _type in types:
+                    # If the type is a type imported from typing
+                    # like Tuple, List, Dict then only the type part
+                    # is extracted and appended to _all_type.
+                    # TODO: use builtin function in typing to extract the typename.
+                    if "typing" in str(_type):
+                        _all_type += str(_type).split(".")[1] + ","
+                    else:
+                        _all_type += _type.__name__ + ','
+                _all_type = _all_type.lstrip(" ")  # Remove any trailing spaces
+
+                if len(types) > 1:
+                    all_args[arg] = {'Union[' + _all_type[:-1] + ']'}
+                else:
+                    all_args[arg] = {_all_type[:-1]}
+            return all_args
```

Comment:
Could you address this comment? thanks

https://github.com/pytorch/pytorch/pull/54513#discussion_r615340718

Reply:
Now addressed.